config/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/client/OAuth2LoginConfigurer.java

```diff
@@ -503,9 +503,14 @@ private AuthenticationEntryPoint getLoginEntryPoint(B http, String providerLogin
 				new OrRequestMatcher(loginPageMatcher, faviconMatcher), defaultEntryPointMatcher);
 		RequestMatcher notXRequestedWith = new NegatedRequestMatcher(
 				new RequestHeaderRequestMatcher("X-Requested-With", "XMLHttpRequest"));
+		RequestMatcher formLoginNotEnabled = (request) -> {
+			DefaultLoginPageGeneratingFilter defaultLoginPageGeneratingFilter = http
+					.getSharedObject(DefaultLoginPageGeneratingFilter.class);
+			return defaultLoginPageGeneratingFilter == null || !defaultLoginPageGeneratingFilter.isFormLoginEnabled();
```

Comment:
Is there another way to determine `isFormLoginEnabled()` without exposing that new method? Ideally we want to patch this to older branches.

Reply:
I don't believe so. At least not while using the `DefaultLoginPageGeneratingFilter`. If we rely on the `isEnabled()` alone it breaks existing tests because it changes redirect behavior with other combinations including just regular oauth2 with a single client. I'm not sure off hand another way to share this type of information with configurers, but I'll look into it.

Reply:
I've updated the PR to utilize `ReflectionUtils` as a way to work around this. Let me know your thoughts, @jgrandja.